pkg/apis/garden/validation/validation.go

```diff
@@ -2026,13 +2025,28 @@ func ValidateBackupInfrastructure(backupInfrastructure *garden.BackupInfrastruct
 func ValidateBackupInfrastructureUpdate(newBackupInfrastructure, oldBackupInfrastructure *garden.BackupInfrastructure) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newBackupInfrastructure.ObjectMeta, &oldBackupInfrastructure.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateBackupInfrastructureMetaUpdate(&newBackupInfrastructure.ObjectMeta, &oldBackupInfrastructure.ObjectMeta,field.NewPath("metadata"))...)
 	allErrs = append(allErrs, ValidateBackupInfrastructureSpecUpdate(&newBackupInfrastructure.Spec, &oldBackupInfrastructure.Spec, newBackupInfrastructure.DeletionTimestamp != nil, field.NewPath("spec"))...)
 	allErrs = append(allErrs, ValidateBackupInfrastructure(newBackupInfrastructure)...)
 
 	return allErrs
 }
 
+// ValidateBackupInfrastructureMetaUpdate validates the updates in metadata of a BackupInfrastructure object.
+func ValidateBackupInfrastructureMetaUpdate(newMeta, oldMeta *metav1.ObjectMeta, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(newMeta, oldMeta, fldPath)...)
+
+	oldPresent, _:= strconv.ParseBool(oldMeta.Annotations[common.BackupInfrastructureForceDeletion])
```

Comment:
Shouldn't we check the error here? The input is coming from a user.

Reply:
Yes. But parseBool or in general we treat non-boolean / errorneous string to be false. So, i thought ignoring error is fine here, WDYT?

Reply:
Agreed!